plugins/modules/docker_container.py

```diff
@@ -663,6 +663,13 @@
       - If I(container_default_behavior) is set to C(compatiblity) (the default value), this
         option has a default of C(no).
     type: bool
+  publish_all_ports:
+    description:
+      - Publish all ports to the host.
+      - Any specified port bindings from I(published_ports) will remain intact when C(true).
+    type: bool
+    default: false
```

Comment:
Right now I'm not really happy with the default for `publish_all_ports`. This is only partially a problem with this PR though, since that behavior was already there before.

The problem is that right now, if the user does not specify `published_ports`, the published ports are not checked for idempotence - **but** the `publish_all_ports` flag is! (If a container has it set, and `docker_container`'s `published_ports` isn't specified, the container will be recreated since there's a config mismatch!) I would argue that the current behavior is a bug, but potentially people depend on it...

The *correct* behavior would be (before this PR) to set `publish_all_ports = None`, and only set it to `True`/`False` if `published_ports` is passed to the module. With the new option it get's a bit more complicated, but it should also not be a problem.

The main problem is what do we do about the old problem. I'm currently tending to treat it as a bug and simply fix it. This changes behavior, but since it prevents container recreation in some situations, I think it's more acceptable than it would be the other way around. What do you think?

Reply:
I tend to agree with you. Any predictable side-effect could be misunderstood as a feature, but the module already has a feature (`recreate`) to let users force recreation in a more obvious way if desired. In other words if they really needed the container to be recreated they should use the intended option otherwise the module is falsely reporting config changes which is a bug.